packages/use-codemirror/src/hooks/useCodeMirror.ts

```diff
@@ -283,11 +298,25 @@ function getCodeMirrorExtensions({
   if (!withoutTheme) extensions.push(customTheme)
 
   // Read only
-  if (readOnly) extensions.push(EditorView.editable.of(false))
+  if (readOnly) {
+    extensions.push(EditorView.editable.of(false))
+  } else {
+    extensions.push(
+      indentOnInput(),
+      bracketMatching(),
+      autocompletion(),
+      closeBrackets(),
+      keymap.of([
+        ...completionKeymap,
+        ...closeBracketsKeymap,
+        selectAllKeyBinding,
+      ]),
```

Comment:
What are these doing? I can’t see any indent, bracket, autocomplete … magic when I play around with the editor. 🤔

Reply:
maybe do a fresh build?